src/components/Microsoft.DotNet.UpgradeAssistant/Checks/WcfServerCheck.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Checks
+{
+    /// <summary>
+    /// Users running upgrade-assistant on projects that build WCF Services
+    /// should be informed that this is an unsupported technology before they
+    /// invest significant time running the tool.
+    /// </summary>
+    public class WcfServerCheck : BypassableReadinessCheck
+    {
+        public WcfServerCheck(ILogger<WcfServerCheck> logger, UpgradeOptions upgradeOptions)
+            : base(logger, upgradeOptions)
+        {
+        }
+
+        /// <summary>
+        /// Gets the value WebFormsCheck.
+        /// </summary>
+        public override string Id => nameof(WcfServerCheck);
+
+        /// <summary>
+        /// Gets a value indicating whether this check can be bypassed using the ReadinessAcknowledgement option.
+        /// </summary>
+        public override bool IsBypassable => true;
+
+        public override string TechnologyDetected => "WCF Server-side Services";
+
+        // TODO: code review - do we expect people to copy and paste this? Maybe we need to make a friendly URL.
+        public override string SupportLink => "https://docs.microsoft.com/en-us/dotnet/architecture/porting-existing-aspnet-apps/migrate-web-forms";
+
+        protected override Task<bool> DoesProjectContainTechnologyAsync(IProject project, CancellationToken token)
+        {
+            if (project is null)
+            {
+                // this is not a valid scenario
+                // satisfies CA1062
+                return Task.FromResult(false);
+            }
+
+            // are there any svc.cs files in this project?
+            return Task.FromResult(project.FindFiles($".svc{GetExtensionForLanguage(project)}").Any());
```

Comment:
There are other ways users might be using WCF. Let's add an issue to track expanding this one to also do some semantic analysis looking for ServiceHost or other key types.

Reply:
#624